xgcm/test/test_grid_ufunc.py

```diff
@@ -511,6 +452,168 @@ def grad_to_inner(a):
+        """
+
+        def diff_center_to_left(a):
+            return a[..., 1:] - a[..., :-1]
+
+        grid = create_1d_test_grid("depth")
+        da = np.sin(grid._ds.depth_c * 2 * np.pi / 9).chunk()
+        da.coords["depth_c"] = grid._ds.depth_c
+
+        diffed = (da - da.roll(depth_c=1, roll_coords=False)).data
+        expected = xr.DataArray(
+            diffed, dims=["depth_g"], coords={"depth_g": grid._ds.depth_g}
+        ).compute()
+
+        # Test direct application
+        result = apply_as_grid_ufunc(
+            diff_center_to_left,
+            da,
+            axis=[("depth",)],
+            grid=grid,
+            signature="(X:center)->(X:left)",
+            boundary_width={"X": (1, 0)},
+            dask="parallelized",
+        ).compute()
+        assert_equal(result, expected)
+
+        # Test Grid method
+        result = grid.apply_as_grid_ufunc(
+            diff_center_to_left,
+            da,
+            axis=[("depth",)],
+            signature="(X:center)->(X:left)",
+            boundary_width={"X": (1, 0)},
+            dask="parallelized",
+        )
+        assert_equal(result, expected)
+
+        # Test decorator
+        @as_grid_ufunc(
+            "(X:center)->(X:left)",
+            boundary_width={"X": (1, 0)},
+            dask="parallelized",
+        )
+        def diff_center_to_left(a):
+            return a[..., 1:] - a[..., :-1]
+
+        result = diff_center_to_left(
+            grid,
+            da,
+            axis=[("depth",)],
+        ).compute()
+        assert_equal(result, expected)
+
+    def test_2d_padding(self):
+        def diff(a, axis):
+            def _diff(a):
+                return a[..., 1:] - a[..., :-1]
+
+            return np.apply_along_axis(_diff, axis, a)
+
+        def vort(u, v):
+            """This needs to return an array 1 element smaller along both axis -1 & -2."""
+
+            u_trimmed = u[..., 1:]
+            v_trimmed = v[..., 1:, :]
+
+            result = diff(v_trimmed, axis=-1) - diff(u_trimmed, axis=-2)
+            return result
+
+        grid = create_2d_test_grid("lon", "lat")
+
+        U = grid._ds.lon_g ** 2 + grid._ds.lat_c ** 3
+        V = grid._ds.lon_c ** 3 + grid._ds.lat_g ** 2
+
+        diffed_v = (V - V.roll(lon_c=1, roll_coords=False)).data
+        diffed_u = (U - U.roll(lat_c=1, roll_coords=False)).data
+        expected = xr.DataArray(
+            diffed_v - diffed_u,
+            dims=["lon_g", "lat_g"],
+            coords={"lon_g": grid._ds.lon_g, "lat_g": grid._ds.lat_g},
+        ).compute()
+
+        result = grid.apply_as_grid_ufunc(
+            vort,
+            U,
+            V,
+            axis=2 * [("lon", "lat")],
+            signature="(lon:left,lat:center),(lon:center,lat:left)->(lon:left,lat:left)",
+            boundary_width={"lon": (1, 0), "lat": (1, 0)},
+            dask="parallelized",  # data isn't chunked along lat/lon
+        )
+        assert_equal(result, expected)
```

Comment:
This test is failing with different values in the assert and I'm not yet sure if it's my result or expected DataArray that is incorrect.